src/main/java/com/morlunk/jumble/audio/AudioOutput.java

```diff
@@ -189,15 +189,28 @@ private boolean mix(short[] outBuffer) {
             }
 
             if(!mMixBuffer.isEmpty()) {
+
+                /* Declare temp vars here to avoid creation/destructions within loops */
+                short pcm = 0;
+                int pcm_tmp = 0;
+
                 for(AudioOutputSpeech.Result result : mMixBuffer) {
                     float[] buffer = result.getSamples();
                     for(int i = 0; i < BUFFER_SIZE; i++) {
-                        short pcm = (short) (buffer[i]*Short.MAX_VALUE); // Convert float to short
-                        pcm = pcm <= Short.MAX_VALUE ? (pcm >= Short.MIN_VALUE ? pcm : Short.MIN_VALUE) : Short.MIN_VALUE; // Clip audio
-                        outBuffer[i] += pcm;
+
+                        /* @TODO replace with higher quality mixing algorithm (better quality and maintaining loudness) */
+
+                        /* Convert float to short. float pcm values are in range [-1.0,1.0] */
+                        pcm = (short) (buffer[i]*Short.MAX_VALUE);
```

Comment:
Overflow here is not possible as it is explicitly down cast which just cuts off at min or max short. PCM values are in range [-1.0,1.0] otherwise they have not been correctly normalized, dynamically compressed or clipped. If it exceeds the ranges then it will be cut anyway. Doing all the calculation in float costs a lot of performance. And this only because of clipping a short is not worth it. This is anyway only the fix for this primitive clipping. As said, I will integrate a much smoother algorithm with a so called "soft knee" (see https://en.wikipedia.org/wiki/Dynamic_range_compression) which has been developed by Paul Vögler.